lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java

```diff
@@ -1191,6 +1206,56 @@ private Task claim(String taskId, boolean forceClaim)
     return task;
   }
 
+  private Task requestReview(String taskId, boolean force)
+      throws TaskNotFoundException, NotAuthorizedException, InvalidTaskStateException,
+          InvalidOwnerException {
+    String userId = taskanaEngine.getEngine().getCurrentUserContext().getUserid();
+    TaskImpl task;
+    try {
+      taskanaEngine.openConnection();
+      task = (TaskImpl) getTask(taskId);
+
+      TaskImpl oldTask = task.copy();
+      oldTask.setId(taskId);
+      oldTask.setExternalId(task.getExternalId());
+      oldTask.setAttachments(task.getAttachments());
+      oldTask.setSecondaryObjectReferences(task.getSecondaryObjectReferences());
+
+      if (!force && !task.getState().equals(TaskState.CLAIMED)) {
+        throw new InvalidTaskStateException(task.getId(), task.getState(), TaskState.CLAIMED);
+      }
+      if (!force && !task.getOwner().equals(userId)) {
+        throw new InvalidOwnerException(userId, task.getId());
+      }
+      if (task.getState().isEndState()) {
+        throw new InvalidTaskStateException(
+            task.getId(), task.getState(), TaskState.CLAIMED, TaskState.READY);
```

Comment:
You only verify that the task is not an end state, but the error message requires CLAIMED and READY.
If we want to allow all states (except for end states) you can use `EnumUtils.allValuesExceptFor(TaskState.END_STATES)`

Reply:
Ok :)